### ISSUE TITLE
In an array type switch, 'def' is not correctly parsed

### ISSUE THREAD

**Issue by GHNewbiee** [szaghi/FLAP#103]
When a switch is a type of **array**, then `def` value should be an **array**, too. For the time being, only the first element is considered, which is not right; unless I miss something. *Tia*

*Example*

```fortran
call cli%add(                    &
  switch     = '--display',      &
  switch_ab  = '-dis',           &
  help       = '...',            &
  help_color = 'cyan',           &
  help_style = 'bold_on',        &
  required   = .false.,          &
  act        = 'store',          &
  nargs      = '+',              &
  def        = '1,2,4,40,21,25', &
  choices    = '...'             &
)

call cli%get_varying(switch='-dis', val=display, error=error)

print *, size(display), display !< prints 1 1
```

**Comment by GHNewbiee** [comment]
Sorry, `def` definitely is an **array**. It's just of size 1, because (I suspect) string is not parsed in a correct manner. *Ta*

**Comment by szaghi** [comment]
Hi,

As a design choice, I decided that the default value (as well as the value passed by command line, if any) is always as a string, no matter of what it is really (scalar, array, real, integer...). When the user invoke any "get" method, the string is casted to the correct type/rank. 

In your example I think that the problem is due to the array values separator: by default it is a white space, not a comma. Indeed, it should be customizable, but the documentation is poor, my bad.

I'll try to update the documents soon, I have introduced new important features in the incoming release.

Kind regards
